### PR TITLE
GCW-3186-Adds quantity changes in requested_packages

### PR DESCRIPTION
**Pull request by shreyas098**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-3186

### What does this PR do?

FEATURE: Adds quantity changes in requested_packages

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -309,10 +309,11 @@ def package_abilities
              search_stockit_items remove_from_set designate register_quantity_change
              mark_missing move print_inventory_label stockit_item_details
              split_package add_remove_item contained_packages parent_containers
-             fetch_added_quantity], Package
+             fetch_added_quantity requested_packages], Package
```
Comment (bharat619) [inline app/models/ability.rb:312]: - remove the requested_packages from both places
- add it in the requested_packages_abilities as
```ruby
def requested_packages_abilities
    can %i[create destroy index checkout update], RequestedPackage, user_id: @user_id
  end
```

Comment (shreyas098) [reply app/models/ability.rb:312]: 👍

File: app/controllers/api/v1/requested_packages_controller.rb [hunk 2]
```diff
@@ -32,12 +33,19 @@ def index
 
       api :POST, '/v1/requested_packages', "Create a cart item"
       def create
-        if params[:quantity].blank?
-          @requested_package.quantity = PackagesInventory::Computer.available_quantity_of(@requested_package.package)
-        end
```
Comment (patrixr) [inline app/controllers/api/v1/requested_packages_controller.rb:37]: Can you add a test case to make sure the api handles missing params correctly ? : )

Comment (shreyas098) [reply app/controllers/api/v1/requested_packages_controller.rb:37]: sure

**Pull request merged**